### PR TITLE
Change key

### PR DESCRIPTION
**Pull request by raleighgreen**
Hi Kevin,

I've made some changes to the code and it seems to be working pretty well.
The `getScale()` function now maps the given scale to all six strings in any key.
I'm wondering if this seems like a good way to have gotten it working?
If so, should the new code be refactored and what would be a good approach for that?

About to send out my first pull request - hoping it goes through ok.
Thanks Kevin!

### CODE REVIEW

**Comment by kmcgillivray** [comment]
Hey @raleighgreen, just wanted to let you know that I see this pull request and I will do a review on GitHub! This will definitely be much easier to manage than Bloc messages.

**Comment by raleighgreen** [comment]
Sounds great Kevin, thanks so much!

rg

**Comment by kmcgillivray** [comment]
Hey Raleigh,

Let's break this down a bit. There are two main goals:
1. Organize the _data_ that represents the guitar in a logical way.
2. Write algorithms that can search and manipulate that data to give us the information we want (a scale in this case).

Let's focus on the first item. This is really a data organization problem, not a step-by-step programming problem. The question is, what's the best way to organize the data so that it's flexible and best represents a real guitar? We can start this organization process by thinking about what different "objects" are involved.
1. **Notes** – The individual tones produced by the guitar from low to high. One note for each tone between the guitar's lowest pitch and highest pitch. Each note might show up more than once on the fretboard, but we just want one object to represent each unique tone!
2. **Strings** – The individual strings that have their own low to high range as a subset of the range of the guitar.
3. **Frets** – The individual places on a string. Each fret has one note and belongs to one string. Multiple frets can have the same note. This is an important distinction, frets and notes are not the same thing!

We could also consider the entire guitar as an object, but for the sake of simplicity let's leave that out for now.

Now we can start to describe each of these types of objects in terms of their properties.
1. A **Note** has a name and a pitch (represented by an audio file in our case). We've discovered that it's easier to keep track of notes by an ID number rather than their letter name because the letter name changes depending on the key.
2. A **String** has a name (E A D G B E) and a low range and a high range.
3. A **Fret** has a Note and a String that it belongs to.

I bet you could come up with a cool way to represent this visually... there are formalized methods for sketching out objects like this, and it's often the first step in designing complex software! Here's a rough outline of the properties and the type of data they hold:

**Note**
id: number
audioFile: string

**String**
name: string (as in a string of letters in JavaScript, not the string itself!)
lowNote: number
highNote: number

**Fret**
note: object
string: object

You'll notice that the Fret's properties are other objects! That means we can attach Note and String objects directly to the fret. The cool thing about this is that objects are passed by "reference" which means that they aren't copied every time they're passed to a function or attached to an object (which would double, triple, or quadruple memory usage), a reference to the original object is passed around and the script accesses the original object whenever it's mentioned.

Basically what we want to do is generate an object for each note. Then generate an object for each string. Then generate an object for each fret that references the correct note and string.

Check out [this JSBin](http://jsbin.com/cilatazaja/1/edit?js,console) for an example of how this might be implemented (it has lots of comments explaining this process again). Once the data is organized, it will be much easier to write the functions we need to calculate scales. This might feel like taking a step backward, but really it's just revising a previous rough draft!

**Comment by raleighgreen** [comment]
Kevin,

Thanks so much for this breakdown.
I'll be going over this thoroughly!
Much appreciated and have a great weekend.

rg
